Use PSA more often in `pk_verify_ext()`

mpg opened the pull request:
Fixes https://github.com/Mbed-TLS/mbedtls/issues/5277 - going with option 2 as it's such a small difference with option 1.

## Gatekeeper checklist

- [x] **changelog** not required
- [x] **backport** no - development only
- [x] **tests** provided

mprse reviewed (commented):
This PR correctly implements option 2 (which is consistent with the PR description) from https://github.com/Mbed-TLS/mbedtls/issues/5277.
I'm not sure if this is ok as in the discussion option 1 was selected by everyone. 

Found only one minor style issue.

mprse commented:
Also this PR requires rebase to resolve conflicts.

mpg commented:
@mprse Thanks for your review. I rebase to resolve the conflict and fix the style issue while at it.

@yanesca @gilles-peskine-arm Since you participated in the initial discussion, can one of you review this PR and confirm if you're OK with it implementing option 2 even though we agreed on option 1? It was only two more lines of code, so I felt it was OK. Thanks in advance!

yanesca commented on tests/suites/test_suite_x509parse.data:
```diff
@@ -784,7 +784,7 @@ depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_X509_RSASSA_PSS_SUPPORT:MBEDTLS_HAS_ALG_S
 
 X509 CRT verification #68 (RSASSA-PSS, wrong salt_len)
-depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_X509_RSASSA_PSS_SUPPORT:MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_HAS_ALG_SHA_1_VIA_MD_OR_PSA_BASED_ON_USE_PSA
+depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_X509_RSASSA_PSS_SUPPORT:MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_HAS_ALG_SHA_1_VIA_MD_OR_PSA_BASED_ON_USE_PSA:!MBEDTLS_USE_PSA_CRYPTO
```
I think we could have a similar test case depending on `MBEDTLS_USE_PSA_CRYPTO` and expecting to pass.

yanesca replied:
And maybe a third one where the salt_len is wrong, but we expect failure whether PSA is being used or not. (If we don't have such test already.)

mpg replied:
Good point, I'll do that!

yanesca reviewed (approved):
Have a minor suggestion, but other than that it looks good to me. I am happy with option 2 as well.

mpg commented:
>  implementing option 2 even though we agreed on option 1? It was only two more lines of code, so I felt it was OK.

Actually it was not just two lines of code, it was one more test (with test data that's pretty hard to generate) and one more special case to explain in the documentation... "the function usually checks the salt len, except if `USE_PSA` is enabled, except in that case it still checks when the salt len happens to be equal to the hash len"... we really want to avoid nested exceptions...

So I changed my mind again and went with option 1, especially as it's what we agreed on initially. This time with complete tests and documentation too. Also rebased on top of development to fix a conflict that appeared in the meantime.

mpg commented:
Also, I'm not sure if this needs a ChangeLog entry: it's not really user-visible, so I'm inclined to say no, please tell if you disagree.

yanesca reviewed (approved):
Looks good to me. (Approval subject to the CI passing.)

mprse reviewed (dismissed):
LGTM.
Restarted CI.

mprse commented:
It looks like there is still a problem:
```

depends.py: !MBEDTLS_X509_RSASSA_PSS_SUPPORT PASSED
depends.py: !MBEDTLS_ECDSA_C !MBEDTLS_ECP_C !MBEDTLS_PKCS1_V21 !MBEDTLS_RSA_C !MBEDTLS_X509_RSASSA_PSS_SUPPORT passed
depends.py: !MBEDTLS_PKCS1_V15 FAILED
^^^^test_depends_py_pkalgs_psa: test/build: depends.py pkalgs (gcc) with MBEDTLS_USE_PSA_CRYPTO defined: tests/scripts/depends.py pkalgs -> 1^^^^
```

mprse reviewed (approved):
Issue on CI is solved.
LGTM
